packages/db/src/query/device.ts

```diff
@@ -168,3 +168,67 @@ async function copyToHistoryTable(
         powerEstimation: device.powerEstimation,
     });
 }
+
+export async function saveDeviceToPeakDb(sensorDataSequenceId: string, deviceName: string, userId: string) {
+    try {
+        const deviceResult = await db
+            .select()
+            .from(device)
+            .where(and(eq(device.name, deviceName), eq(device.userId, userId)))
+            .limit(1);
+
+        if (deviceResult.length === 0) {
+            throw new Error("Gerät nicht gefunden.");
+        }
+
+        await db.insert(deviceToPeak).values({
+            deviceId: deviceResult[0].id,
+            sensorDataSequenceId,
+        });
+    } catch (error) {
+        console.error("Error saving device to peak:", error);
+        throw new Error("Failed to save device to peak.");
+    }
+}
+
+export async function createStandardDevicesIfNotExist(userId: string) {
+    const existingDevices = await getDevicesByUser(userId);
+    const existingDeviceNames = new Set(existingDevices.map((device) => device.name));
+
+    const deletedDevices = await db.select().from(deviceHistory).where(eq(deviceHistory.userId, userId));
```

Comment:
Eigentlich werden auch bei Änderungen Einträge in die deviceHistory Tabelle geschrieben. Es kann also sein, dass da ein noch existentes Gerät drin steht. In dem Fall funktioniert die Logik nicht